### PR TITLE
(WIP) Free text advance search.

### PR DESCRIPTION
**Pull request by ravi-sharma**

### CODE REVIEW

File: CRM/Activity/DAO/Activity.php [hunk 1]
```diff
@@ -234,6 +234,13 @@ static function &fields() {
           'dataPattern' => '',
           'export' => true,
         ) ,
+        'activity_free_text_search' => array(
```
Comment (seamuslee001) [inline CRM/Activity/DAO/Activity.php]: Not sure why this or the code below is in the DAO, should be in the BAO class

Comment (ravi-sharma) [inline CRM/Activity/DAO/Activity.php]: @seamuslee001 Made few adjustments, could please please check again ?

Comment (seamuslee001) [inline CRM/Activity/DAO/Activity.php]: Hi @ravi-sharma what i was meaning is i think this added code and the getFreeTextFields() need to move into https://github.com/civicrm/civicrm-core/blob/master/CRM/Activity/BAO/Activity.php and then called from there, Unless you are actually planning on adding a column to the civicrm_activity table this is the wrong place. 

The DAO class is designed to be just about mimicking the database, these get regenerated whenever the schema changes. Most code should either go in the form layer or in the BAO class which is for the Business logic

Comment (ravi-sharma) [inline CRM/Activity/DAO/Activity.php]: Hi @seamuslee001 Sorry for the confusion, just made the adjustments. please check now.

**Comment by colemanw** [comment]
@civicrm-builder add to whitelist

**Comment by eileenmcnaughton** [comment]
I'm going to close this on the basis that it is stale & it would be better for you to fix that & re-open an new PR than to leave this stale. Administratively it also needs a JIRA issue.

I'm kind of inclined to think this might be better in an extension than in core

**Pull request closed without merging**